extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -22,19 +22,77 @@ import { AcctStore } from '../../../../js/common/platform/store/acct-store.js';
 
-  public sendableMsg = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingPrv?: Key): Promise<SendableMsg> => {
+  public sendableMsgs = async (newMsg: NewMsgData, pubkeys: PubkeyResult[], signingKey?: ParsedKeyInfo): Promise<MultipleMessages> => {
     if (newMsg.pwd && !this.isDraft) {
       // password-protected message, temporarily uploaded (already encrypted) to:
       //    - flowcrypt.com/api (consumers and customers without on-prem setup), or
       //    - FlowCrypt Enterprise Server (enterprise customers with on-prem setup)
       //    It will be served to recipient through web
-      const { url: msgUrl, externalId } = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
+      const uploadedMessageData = await this.prepareAndUploadPwdEncryptedMsg(newMsg); // encrypted for pwd only, pubkeys ignored
+      // pwdRecipients that have their personal link
+      const pwdRecipients = Object.keys(uploadedMessageData.emailToExternalIdAndUrl ?? {}).filter(email => !pubkeys.some(p => p.email === email));
       newMsg.pwd = undefined;
-      return await this.sendablePwdMsg(newMsg, pubkeys, { msgUrl, externalId }, signingPrv); // encrypted for pubkeys only, pwd ignored
-    } else if (this.richtext) { // rich text: PGP/MIME - https://tools.ietf.org/html/rfc3156#section-4
+      const collectedAttachments = await this.view.attachmentsModule.attachment.collectEncryptAttachments(pubkeys);
+      const pubkeyRecipients: { [type in RecipientType]?: EmailParts[] } = {};
+      for (const [key, value] of Object.entries(newMsg.recipients)) {
+        if (['to', 'cc', 'bcc'].includes(key)) {
+          const sendingType = key as RecipientType;
+          pubkeyRecipients[sendingType] = value?.filter(emailPart => pubkeys.some(p => p.email === emailPart.email)
+            // pwd recipients that have no personal links go to legacy message
+            || (uploadedMessageData.emailToExternalIdAndUrl || {})[emailPart.email] === undefined);
+        }
+      }
+      const uniquePubkeyRecipientToAndCCs = Value.arr.unique((pubkeyRecipients.to || []).concat(pubkeyRecipients.cc || [])
+        .map(recipient => recipient.email.toLowerCase()));
+      // pubkey recipients should be able to reply to "to" and "cc" pwd recipients
+      const replyTo = (newMsg.recipients.to ?? []).concat(newMsg.recipients.cc ?? [])
+        .filter(recipient => !uniquePubkeyRecipientToAndCCs.includes(recipient.email.toLowerCase()));
+      const msgs: SendableMsg[] = [];
+      if (pubkeyRecipients.to?.length || pubkeyRecipients.cc?.length || pubkeyRecipients.bcc?.length) {
+        const pubkeyMsgData = {
+          ...newMsg,
+          recipients: pubkeyRecipients,
+          // brackets are required for test emails like '@test:8001'
+          replyTo: replyTo.length ? `${Str.formatEmailList([newMsg.from, ...replyTo], true)}` : undefined
+        };
+        msgs.push(await this.sendablePwdMsg(
```

Comment:
I introduced a method named `sendablePubkeyMsgWithPwdLink` to preserve the original behaviour.
It format message body slightly differently (in case of `!richtext`) and also embeds the legacy link.

Reply:
Now, `sendableNonPwdMsg`, `sendablePwdMsg` and `sendablePubkeyMsgWithPwdLink` are the main points, 
`sendableSimpleTextMsg` and `sendableRichTextMsg` are deleted.